pypdfocr/pypdfocr_tesseract.py

```diff
@@ -91,37 +92,9 @@ def _is_version_uptodate(self):
         for line in ret_output.splitlines():
             if 'tesseract' in line:
                 ver_str = line.split(' ')[1]
-                if ver_str.endswith('dev'): # Fix for version strings that end in 'dev'
-                    ver_str = ver_str[:-3]
-
-        # Iterate through the version dots
-        ver = [int(x) for x in ver_str.split('.')]
-        req = [int(x) for x in self.required.split('.')]
-
-        # Aargh, in windows 3.02.02 is reported as version 3.02  
-        # SFKM
-        if str(os.name) == 'nt':
-            req = req[:2]
-
-        version_good = False
-        for i,num in enumerate(req):
-            if len(ver) < i+1:
-                # This minor version number is not present in tesseract, so it must be
-                # lower than required.  (3.02 < 3.02.01)
-                break
-            if ver[i]==num and len(ver) == i+1 and len(ver)==len(req):
-                # 3.02.02 == 3.02.02
-                version_good = True
-                continue
-            if ver[i]>num:
-                # 4.0 > 3.02.02
-                # 3.03.02 > 3.02.02
-                version_good = True
-                break
-            if ver[i]<num:
-                # 3.01.02 < 3.02.02
-                break
-            
+
+        version_good = version.parse(self.required) < version.parse(ver_str)
```

Comment:
Should this not be less than or equal, rather than strictly less than?
Also, does it cope with nt using shorter version numbers?
These are both covered in the tests, so should be easy to check.

Reply:
Both of your points are valid and both fail on tests - even if I put <= there, it fails for 3.02.02 and 3.02.02 - which I think is rather strange.
Guess I have to rework that. Thanks for pointing it out.

Reply:
That is strange. It's a shame as this is a nice solution.

I used `parse_version` from `pkg_resources` (https://github.com/benjsec/pypdfocr/commit/d8a6067fc20f78f27468ad2c2a1a4f7d6d22148a#diff-e56576c0e33cf639c81a882b60ea2f0e) which is somewhat less neat. Also hasn't been tested with `4.00.00alpha` yet, I'll add another unit test when I get chance.

Reply:
The `runtest.py` seems to use pre-compiled code, so I've been bypassing it for my own work. I've just been using pytest directly (as `pytest test`), which should give you more output.

Reply:
Oh dear, now I feel embarrassed - it was always testing against my installed version of 0.9.1, no wonder it didn't change at all.
I've changed it a bit for the test with windows and added some test cases for dev and alpha:

[pypdfocr_tesseract.zip](https://github.com/virantha/pypdfocr/files/1827250/pypdfocr_tesseract.zip)

I'm not so happy with the way I solved the Windows reporting less version information, as it just removes the last 3 chars from the string, so as it is, it would require the requirement to always be given as x.yy.zz - that's not very intelligent now.